README.md

```diff
@@ -10,10 +21,73 @@ Error code  | Error Name | Description
  1   | Server Exception | Unknown error from the server 
  2   | Bad request | The request make do not follow expected payload 
 
-# Messages
+# Handshake
+When a client connect to a Lama node. It need to init a handshake, the sequence of messages does not evolve.
+
+## Sequence
+Client => Server : Authentification
+Server => Client : Supported protocol version
+Client => Server : Choose a protocol version to use
+
+## Authentification
+
+Client call the server, for authentification.
+```
+auth_method => UINT8
+identity_name => STRING
+identity_key => [BUFFER]
+```
+
+Field | Description
+------|--------------
+auth_method | Version of the auth method (always 0 currently).
+identity_name | The name of the client
+identity_key | Optional buffer containing a key for identifily uniquely a client
+
+## Supported protocol version
+Server answer (if success) with the protocol version supported
+
+```
+max_version_major_supported=UINT8
+max_version_minor_supported=UNIT8
+min_version_major_supported=UINT8
+min_version_minor_supported=UINT8
+```
+
+Field | Description
+-------|-----------
+max_version_major_supported | Max version supported (major.minor.patch) of the protocol
+max_version_minor_suppored | Max version supported (major.minor.patch) of the protocol
+min_version_major_supported | Min version supported (major.minor.patch) of the protocol
+min_version_minor_supported | Min version supported (major.minor.patch) of the protocol
+
+## Choose the protocol version to use
+Client answer
+```
+version_used_major=UINT8
+version_used_minor=UINT8
+```
+
+
+# Messages
 
  Message Code | Message Name | Description 
 -------------|---------------|--------------
-1 | something | ok
+1 | Push message | Push a message on a given server or client
+2 | 
 
+## Push message
 
+Push a message on a server
+```
+message type => UINT8
+created at => UINT32
+routing key => STRING
+headers => ARRAY (header)
+header{
+  key => STRING
+  value => STRING
+}
```

Comment:
At start, I want to use JSON for routing key & additional header, but I think, finaly, JSON.parse, it's fucking slow... this approch could be lot of faster, and keep the possibility of letting people extend the protocol